### PR TITLE
Currency Exchange API Contract

### PR DESCRIPTION
**Pull request by SBagaria2710**

### CODE REVIEW

**Review by harshith-venkatesh — DISMISSED** [review]
LGTM

**Comment by harshith-venkatesh** [comment]
@Akashdeep-Patra ,if you are good with the API contract, we can merge it

**Comment by Akashdeep-Patra** [comment]
Looks just fine

**Review by Akashdeep-Patra — APPROVED** [review]

File: crypto/currency-exchange/README.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+}
+```
+
+## Currency Exchange Object
+
+```
+{
+    from: {
+        "id": <currency_id>,
+        "name": <currency_name>
```
Comment (ankushdharkar) [inline crypto/currency-exchange/README.md:24]: Why do we need to repeat the name here, can we not extract this from the currency id?

File: crypto/currency-exchange/README.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    },
+    to: {
+        "id": <currency_id>,
+        "name": <currency_name>
+    },
+    value: <exchange_value>
```
Comment (ankushdharkar) [inline crypto/currency-exchange/README.md:30]: Question: Does value implicitly mean 1 'from' = (value) 'to'?

File: crypto/currency-exchange/README.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+```
+
+## Exchange Request Object
+
+```
+{
+  from: <src_currency_id>,
+  to: <trgt_currency_id>,
+  quantity: <selected_quantity>
+}
```
Comment (ankushdharkar) [inline crypto/currency-exchange/README.md:41]: How do we know which user is requesting this exchange? How do we know if it succeeded?

File: crypto/currency-exchange/README.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+## **Requests**
+
+|                         Route                         |                                Description                                 |
+| :---------------------------------------------------: | :------------------------------------------------------------------------: |
+|      [GET /currency-list](#get-currency-listing)      |     Returns all accepted currency with their availability in the bank      |
+| [GET /currency-exchange-rate](#get-currency-exchange) | Returns currency with their current exchange value with the other currency |
+|           [PATCH /convert/](#patch-convert)           |           Update the user wallet and bank currency availability            |
+
+## **GET /currenct-list**
```
Comment (ankushdharkar) [inline crypto/currency-exchange/README.md:52]: Typo?

File: crypto/currency-exchange/README.md [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+}
+```
+
+## **Requests**
+
+|                         Route                         |                                Description                                 |
+| :---------------------------------------------------: | :------------------------------------------------------------------------: |
+|      [GET /currency-list](#get-currency-listing)      |     Returns all accepted currency with their availability in the bank      |
+| [GET /currency-exchange-rate](#get-currency-exchange) | Returns currency with their current exchange value with the other currency |
+|           [PATCH /convert/](#patch-convert)           |           Update the user wallet and bank currency availability            |
```
Comment (ankushdharkar) [inline crypto/currency-exchange/README.md:50]: 'convert' sounds too generic. Can we get a specific name for this like above please?

Comment (ankushdharkar) [reply crypto/currency-exchange/README.md:50]: Should this be a PATCH or a POST?

File: crypto/currency-exchange/README.md [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+- Currency
+  > this will contain details about the accepted currencies at crypto-website
+
+## Currency Object
+
+```
+{
+    "id": <currency_id>,
+    "name": <currency_name>,
+    "quantity": <available_quantity>
```
Comment (ankushdharkar) [inline crypto/currency-exchange/README.md:14]: Why should we tell the client all the availabilities?

**Review by ankushdharkar — CHANGES_REQUESTED** [review]

**Pull request closed without merging**